### ISSUE TITLE
[k8s.io] [HPA] Horizontal pod autoscaling (scale resource: CPU) [k8s.io] [Serial] [Slow] ReplicationController failing with GCI

### ISSUE THREAD

**Issue by vishh** [kubernetes/kubernetes#31928]
This test is failing often on the gci-serial suite both on master and release-1.4.

[Test Grid link](https://k8s-testgrid.appspot.com/google-gci#gci-serial&sort-by-failures=)

GCI is the default in v1.4 and so it is imperative to at-least triage the flakes soon. 

cc @kubernetes/autoscaling

**Comment by vishh** [comment]
cc @piosz @mwielgus @jszczepkowski

**Comment by jszczepkowski** [comment]
This is probably a duplicate of https://github.com/kubernetes/kubernetes/issues/30939 (problems with heapster).

**Comment by vishh** [comment]
Given that the flake is happening pretty consistently, will that help
triaging further? I'm concerned if GCI has any metrics issues.

On Fri, Sep 2, 2016 at 1:52 AM, Jerzy Szczepkowski <notifications@github.com

**Comment by piosz** [comment]
@vishh is this issue about nodes running na GCI?
@timstclair @dchen1107 is there any chance that #30939 might happen on GCI?

**Comment by timstclair** [comment]
Based on [derek's comment](https://github.com/kubernetes/kubernetes/issues/30939#issuecomment-244436150) I don't think it would happen on GCI since they don't appear to use the `init.scope`. @kubernetes/goog-image to confirm.

**Comment by jszczepkowski** [comment]
> This is probably a duplicate of #30939 (problems with heapster).

#30939 was solved and it was not causing this problem.

**Comment by jszczepkowski** [comment]
This is probably a duplicate of #31591.

I cannot verify it as I have problems accessing logs.

**Comment by vishh** [comment]
@jszczepkowski What issues are you having with access to logs?

**Comment by jszczepkowski** [comment]
@vishh 
When I click `Filter through log files` I get
```
Exceeded soft private memory limit of 128 MB with 166 MB after servicing 0 requests total
```

**Comment by vishh** [comment]
That feature hasn't worked ever! You can access logs [here](https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/logs/kubernetes-e2e-gce-gci-serial/66) - essentially click on `artifacts`.

**Comment by jszczepkowski** [comment]
Thanks for help. Still, there are no kernel logs (`kern.log` file). OOM-kill is reported inside them.

Can we try to reproduce it with collection of kernel logs turned on?

**Comment by vishh** [comment]
How do we collect logs in the e2e suite? Do we have support for `journald` in there?

**Comment by vishh** [comment]
Can someone from @kubernetes/sig-node help with ensuring support for journald in e2e log collection?

**Comment by thockin** [comment]
@vishh Can you nominate someone to help?  Crying for help does not work

**Comment by vishh** [comment]
@Random-Liu has looked at journald logs before. @Random-Liu do you have bandwidth to tackle this?
@euank might be able to help as well.

**Comment by jlowdermilk** [comment]
I think this issue is out of date. The HPA rc test is no longer failing on gci-release-1.4 . See [testgrid](https://k8s-testgrid.appspot.com/google-gci#gci-serial&sort-by-failures=)

This test has been green for several days, the consistent failures are for kubelet regular resource usage tracking which is https://github.com/kubernetes/kubernetes/issues/28124, and non-release-blocking iiuc

**Comment by jlowdermilk** [comment]
It does appear tho that this test is what is causing kubernetes-e2e-gke-serial-release-1.4 to timeout ([internal jenkins](http://kubekins.dls.corp.google.com/view/CI%201.4/job/kubernetes-e2e-gke-serial-release-1.4/135/)). It actually looks like a test framework bug. Test is chugging along then appears to lose contact with apiserver.

```
ctrl/proxy/BumpMetric  delta=0&durationSec=30&metric=QPS&requestSizeMetrics=10 }
10:02:21 Sep  8 10:02:21.187: INFO: RC rc: sending request to consume 0 MB
10:02:21 Sep  8 10:02:21.189: INFO: URL: {https  <nil> 104.198.59.115 /api/v1/namespaces/e2e-tests-horizontal-pod-autoscaling-iazb3/services/rc-ctrl/proxy/ConsumeMem  durationSec=30&megabytes=0&requestSizeMegabytes=100 }
10:02:21 Sep  8 10:02:21.550: INFO: RC rc: sending request to consume 375 millicores
10:02:21 Sep  8 10:02:21.552: INFO: URL: {https  <nil> 104.198.59.115 /api/v1/namespaces/e2e-tests-horizontal-pod-autoscaling-iazb3/services/rc-ctrl/proxy/ConsumeCPU  durationSec=30&millicores=375&requestSizeMillicores=20 }
10:02:51 Sep  8 10:02:51.158: INFO: RC rc: sending request to consume 0 of custom metric QPS
10:02:51 Sep  8 10:02:51.194: INFO: RC rc: sending request to consume 0 MB
10:02:51 Sep  8 10:02:51.238: INFO: replicationController: current replicas number 2 waiting to be 3
10:02:51 Sep  8 10:02:51.706: INFO: RC rc: sending request to consume 375 millicores
10:03:07 Sep  8 10:03:07.218: INFO: URL: {https  <nil> 104.198.59.115 /api/v1/namespaces/e2e-tests-horizontal-pod-autoscaling-iazb3/services/rc-ctrl/proxy/ConsumeCPU  durationSec=30&millicores=375&requestSizeMillicores=20 }
10:03:11 Sep  8 10:03:11.205: INFO: Unexpected error occurred: Unable to get server version: the server has asked for the client to provide credentials
10:03:11 Sep  8 10:03:11.240: INFO: Unexpected error occurred: Unable to get server version: the server has asked for the client to provide credentials
10:03:11 Sep  8 10:03:11.245: INFO: replicationController: current replicas number 2 waiting to be 3
10:03:39 Sep  8 10:03:31.255: INFO: replicationController: current replicas number 2 waiting to be 3
10:03:39 Sep  8 10:03:37.389: INFO: RC rc: sending request to consume 375 millicores
10:03:39 Sep  8 10:03:37.392: INFO: URL: {https  <nil> 104.198.59.115 /api/v1/namespaces/e2e-tests-horizontal-pod-autoscaling-iazb3/services/rc-ctrl/proxy/ConsumeCPU  durationSec=30&millicores=375&requestSizeMillicores=20 }
10:04:07 Sep  8 10:04:07.581: INFO: RC rc: sending request to consume 375 millicores
10:04:07 Sep  8 10:04:07.584: INFO: URL: {https  <nil> 104.198.59.115 /api/v1/namespaces/e2e-tests-horizontal-pod-autoscaling-iazb3/services/rc-ctrl/proxy/ConsumeCPU  durationSec=30&millicores=375&requestSizeMillicores=20 }
10:04:11 Sep  8 10:04:11.299: INFO: Unexpected error occurred: an error on the server ("Internal Server Error: \"/api/v1/namespaces/e2e-tests-horizontal-pod-autoscaling-iazb3/replicationcontrollers/rc\"") has prevented the request from succeeding (get replicationControllers rc)
10:04:11 STEP: Removing consuming RC rc
11:40:31 kubernetes-e2e-gke-serial-release-1.4-135
11:40:31 + rc=124
11:40:31 + [[ 124 -eq 124 ]]
11:40:31 
+ echo 'Build timed out'
```

cc @rmmh

**Comment by Random-Liu** [comment]
> @Random-Liu has looked at journald logs before. @Random-Liu do you have bandwidth to tackle this?

Sorry, just seen this. :P 
@euank Thanks a lot for working on this.

**Comment by pwittrock** [comment]
Please apply the `non-release-blocker` label to this issue if it does not need to be fixed before we launch 1.4.  Otherwise please advise on how to proceed with respect to 1.4.

**Comment by jszczepkowski** [comment]
There are two problems here
- not enough requested memory for consumer (2 original problems reported by @vishh) - solved by #31972,
- internal server error, error on the server has prevented the request from succeeding (the new problem reported by @jlowdermilk) - already reported as #32345.

Closing in favor of #32345
